src/main/java/org/csanchez/jenkins/plugins/kubernetes/volumes/DynamicPVC.java

```diff
@@ -0,0 +1,91 @@
+package org.csanchez.jenkins.plugins.kubernetes.volumes;
+
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.OwnerReferenceBuilder;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaim;
+import io.fabric8.kubernetes.api.model.PersistentVolumeClaimBuilder;
+import io.fabric8.kubernetes.api.model.Quantity;
+import io.fabric8.kubernetes.api.model.Volume;
+import io.fabric8.kubernetes.api.model.VolumeBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import java.util.Collections;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import org.apache.commons.lang.StringUtils;
+
+import static java.util.logging.Level.INFO;
+
+/**
+ * Interface containing common code between {@link DynamicPVCVolume} and {@link org.csanchez.jenkins.plugins.kubernetes.volumes.workspace.DynamicPVCWorkspaceVolume}.
+ */
+public interface DynamicPVC {
+    Logger LOGGER = Logger.getLogger(DynamicPVC.class.getName());
+
+    default Volume buildPVC(String volumeName, String podName){
+        String pvcName = getPvcName(podName);
+        return new VolumeBuilder()
+                .withName(volumeName)
+                .withNewPersistentVolumeClaim()
+                .withClaimName(pvcName)
+                .withReadOnly(false)
+                .endPersistentVolumeClaim()
+                .build();
+    }
+
+    default PersistentVolumeClaim createPVC(KubernetesClient client, ObjectMeta podMetaData){
+        String namespace = podMetaData.getNamespace();
+        String podName = podMetaData.getName();
+        LOGGER.log(Level.FINE, "Adding volume from pod: {0}/{1}", new Object[] { namespace, podName });
```

Comment:
Is this `from` or `for`?

Reply:
Rather `for` indeed. I moved the initial code without paying too much attention to the exact wording.